credoai/modules/dataset_modules/dataset_base.py

```diff
@@ -61,4 +63,44 @@ def _make_pipe(self):
                [('scaler', StandardScaler()),
                 ('model', model)])
         return pipe
+
+    def _assess_proxy_in_feature_space(
+        self, excluded_columns=None, method="mutual_information"
+    ):
+        """Assesses features based on how much they serve as a proxy for a sensitive attribute
+        This method relies on features information only (label information is not used)
+
+        Parameters
+        ----------
+        excluded_columns : list[str]
+            List of columns to exclude from the assessment
+        method : str, optional
+            Method to use for quantifying the magnitude of proxy, by default 'mutual_information'
+            Choices:
+                'mutual_information' (the "amount of information" in normalized nats obtained about a feature by observing another feature)
+
+        Returns
+        -------
+        dict
+            Keys are feature names and values are their proxy scores
+            Score between 0.0 and 1.0
+        """
+        proxy_scores = {}
+        
+        if method == "mutual_information":
+            # Exclude features of type float
+            float_cols = list(self.X.select_dtypes(include=["float16", "float32", "float64"]))
+            if excluded_columns:
+                excluded_columns.extend(float_cols)
+            else:
+                excluded_columns = float_cols
+
+            for column in self.X.drop(excluded_columns, axis=1):
+                proxy_scores[column] = normalized_mutual_info_score(
```

Comment:
I'd suggest we use the `feature_selection.mutual_info_clas` ([link](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.mutual_info_classif.html)). It handles continuous and categorical, works across the entire matrix and returns back the information for each feature.

Reply:
Though I see that isn't `adjusted`. Hmmm

Reply:
I still think we should use `mutual_info_classif`. It'll give a relative ordering of variables

Reply:
Calculating mutual information between a continuous and a discrete variable is nontrivial. Only purely continuous or purely discrete pairs are possible. There is [this](https://proceedings.neurips.cc/paper/2017/file/ef72d53990bc4805684c9b61fa64a102-Paper.pdf) NIPS paper that has proposed a solution for mixed cases, but no code and looks advanced. I make updates to return mi for the two possible pairs scenarios.